### ISSUE TITLE
Validation

### ISSUE THREAD

**Issue by ikatun** [rhythnic/react-form-wrapper#47]
Do you have any plans for validation implementation?

Would you be willing to share your plans for implementing the validation and accept PRs? :)

**Comment by rhythnic** [comment]
Yeah that'd be great!  To be honest I haven't used this package in a while and I need to take some time this week to look it over.  I'm working with Vue at work these days and I don't have much time for maintaining it.  Any help would be very welcome.

I'll think about the validation API.  Do you have any thoughts on it?  I like the idea of Form Wrapper calling a handler on every form event and letting the parent component pass back an object where the keys are field paths and the values are anything that the user wants to pass to the input as an error prop, or possibly configurable prop name.  But it might be too expensive or difficult to try to capture all of the form events.  Or it could just call the handler when the data changes, and expose a prop inside the form that allows the developer to selectively capture events.  Or add the events you want to subscribe to in an options object passed to the field function.

**Comment by ikatun** [comment]
Thanks for detailed answer!

My idea was that values accumulation (FormWrapper), event monitoring and value(s) validation can be treated as three separate things, each one exposing its own high order component that can be used on its own or combined with any other.

This allows user to choose the amount of overhead herself :)

I've already written the component for focus/blur capturing and a validation component which accepts an object of validators and injects "valid = path => true/false" method to wrapped component much like your "field" method. Needed them for a project :)

I could share them here with a usage example for discussion?

**Comment by rhythnic** [comment]
Yes!  That would be great.  I like the separate concerns approach.
